File: ruby/planet-time/lib/interplanet_time/formatting.rb
# frozen_string_literal: true

module InterplanetTime
  # Formatting utilities — ported from planet-time.js.
  module Formatting
    extend self

    # Format a light-travel duration (seconds) to human-readable string.
    # Examples: 45 → "45 s", 186 → "3 min 6 s", 3700 → "1 h 1 min 40 s"
    def format_light_time(seconds)
      s   = seconds.round
      h   = s / 3600
      m   = (s % 3600) / 60
      sec = s % 60

      parts = []
      parts << "#{h} h"   if h > 0
      parts << "#{m} min" if m > 0
      parts << "#{sec} s" if sec > 0 || parts.empty?
      parts.join(' ')
    end

    # Format a planet local time as an ISO-8601-like string.
    def format_planet_time_iso(planet, h, m, s)
      hh = h.to_s.rjust(2, '0')
      mm = m.to_s.rjust(2, '0')
      ss = s.to_s.rjust(2, '0')
      "#{hh}:#{mm}:#{ss}+#{planet}"
    end
  end
end

File: go/planet-time/interplanet_time/formatting.go
package interplanet_time

import "fmt"

// FormatLightTime formats a light travel duration in seconds as a human-readable string.
// Examples: 45 → "45 s", 186 → "3 min 6 s", 3700 → "1 h 1 min 40 s"
func FormatLightTime(seconds float64) string {
	s := int(seconds)
	if s < 60 {
		return fmt.Sprintf("%d s", s)
	}
	h := s / 3600
	m := (s % 3600) / 60
	sec := s % 60
	if h > 0 {
		if sec > 0 {
			return fmt.Sprintf("%d h %d min %d s", h, m, sec)
		}
		return fmt.Sprintf("%d h %d min", h, m)
	}
	if sec > 0 {
		return fmt.Sprintf("%d min %d s", m, sec)
	}
	return fmt.Sprintf("%d min", m)
}

// FormatPlanetTimeISO returns a simple ISO-like string for a planet time.
// Format: "planet/HH:MM:SS"
func FormatPlanetTimeISO(planet string, hour, minute, second int) string {
	return fmt.Sprintf("%s/%02d:%02d:%02d", planet, hour, minute, second)
}
